src/middlewared/middlewared/plugins/failover_/datastore.py

```diff
@@ -0,0 +1,125 @@
+import os
+import time
+
+from middlewared.service import CallError, Service
+from middlewared.plugins.config import FREENAS_DATABASE
+from middlewared.plugins.datastore.connection import thread_pool
+from middlewared.utils.threading import start_daemon_thread, set_thread_name
+
+FREENAS_DATABASE_REPLICATED = f'{FREENAS_DATABASE}.replicated'
+
+
+class FailoverDatastoreService(Service):
+
+    class Config:
+        namespace = 'failover.datastore'
+        private = True
+
+    thread_pool = thread_pool
```

Comment:
Am I blind? I dont see where this is used in this class?

Reply:
`Middleware._call_prepare` uses this variable to determine in which thread pool it will launch non-async methods